Reorganizing documentation (#119)
ASUS-AICS/LibMultiLabel: docs/cli/linear.rst

.. code-block:: bash

    python3 main.py --config CONFIG_PATH \
                    --linear \
                    --liblinear_options LIBLINEAR_OPTIONS \
                    --linear_technique MULTILABEL_TECHNIQUE

- **config**: configure parameters in a yaml file.  A validation set is not needed because the program may split the training set for internal validation. If specified, it will be ignored.

The linear classifiers are based on `LIBLINEAR <https://www.csie.ntu.edu.tw/~cjlin/liblinear/>`_, and its options may be specified.

> - **linear**: When this option exists, it is set to True such that the linear classifiers will be run. Otherwise it is set to False by default such that the neural netowrk module will be executed and the program will terminate if the neural network config is not specified.

Reviewer: ```suggestion
- **linear**: If this option exists, it is set to True such that the linear classifiers will be run. Otherwise it is set to False by default such that the neural network module will be executed and the program will terminate if the neural network config is not given.
```

.. code-block:: bash

    python3 main.py --config CONFIG_PATH \
                    --linear \
                    --liblinear_options LIBLINEAR_OPTIONS \
                    --linear_technique MULTILABEL_TECHNIQUE

> - **config**: configure parameters in a yaml file. Notice that the validation set is not needed to be specified additionally as the program will ignore it.

Reviewer: ```suggestion
- **config**: configure parameters in a yaml file.  A validation set is not needed because the program may split the training set for internal validation. If specified, it will be ignored.
```

> The linear classifiers are based on LIBLINEAR, and its options may be specified.

Reviewer: give a link to liblinear page

> - **linear**: When this option exists, it is set to True to tell program that the linear classifier is used. Defaults to False.

Reviewer: I don't think this gives sufficient reasons why this --linear is needed? What if users didn't specify it?

- **liblinear_options**: An `option string for LIBLINEAR <https://github.com/cjlin1/liblinear>`_.

- **linear_technique**: An option for multi-label techniques. We now support ``1vsrest`` (implementing one-vs-rest technique), ``thresholding`` (implementing thresholding technique), and ``cost_sensitive`` (implementing cost-sensitive technique).

.. code-block:: bash

    python3 main.py --config example_config/rcv1/l2svm.yml --linear --linear_technique 1vsrest
    python3 main.py --config example_config/rcv1/l2svm.yml --linear --linear_technique thresholding

> 

Reviewer: why needing examples here? We already have one in the quick start place.
